main.cpp

```diff
@@ -8,14 +8,17 @@ int main(int argc, char *argv[]) {
         return 1;
     }
 
+    std::ifstream file1(argv[1]);
+    std::ifstream file2(argv[2]);
+    if (!file1.good() || !file2.good()) {
```

Comment:
Amíg nem próbáltál olvasni a streamből, addig mindenképp igaz lesz a `good()`. És ez még nem oldja meg a memóriaszivárgást, mert a parsolásnál továbbra is felmerülhetnek hibák. És a hibásan parsolt Unitnak ugyan nem lesz lefoglalva memória, de ha az 1. jó, a 2. meg hibás, akkor az 1.-nek lefoglalt memória elszivárog.